BepInEx.IL2CPP/Preloader.cs

```diff
@@ -83,5 +86,48 @@ public static void Run()
                 throw;
             }
         }
+
+        private static void InitializeUnityVersion()
+        {
+            try
+            {
+                ConfigEntry<String> config = ConfigFile.CoreConfig.Bind(
+                    "Unity", "Version",
+                    String.Empty,
+                    "Allows you to manually set the Unity version .");
+
+                String version = config.Value;
+                if (String.IsNullOrWhiteSpace(version))
+                {
+                    version = //Version.Parse(Application.unityVersion);
+                        Process.GetCurrentProcess().MainModule.FileVersionInfo.FileVersion;
+
+                    PreloaderLogger.Log.LogDebug($"Unity version obtained from main application module: [{version}]");
+                }
+                else
+                {
+                    PreloaderLogger.Log.LogDebug($"Unity version obtained from the config: [{version}]");
+                }
+
+                String[] parts = version.Split('.');
+                Int32 major = 0;
+                Int32 minor = 0;
+                Int32 build = 0;
+
+                // Issue #229 - Don't use Version.Parse("2019.4.16.14703470L&ProductVersion")
+                major = Int32.Parse(parts[0], NumberStyles.Integer, CultureInfo.InvariantCulture);
+                if (parts.Length > 1)
+                    minor = Int32.Parse(parts[1], NumberStyles.Integer, CultureInfo.InvariantCulture);
+                if (parts.Length > 2)
+                    build = Int32.Parse(parts[2], NumberStyles.Integer, CultureInfo.InvariantCulture);
+
+                UnityVersionHandler.Initialize(major, minor, build);
+                Log.LogInfo($"Running under Unity v{major}.{minor}.{build}");
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidDataException("Failed to parse Unity version.", ex);
```

Comment:
I don't think we should rethrow here. You still should be able to at least launch the game without massive exceptions. A better option might be to add a clear Error level log.

Reply:
Why? If we initialize the wrong version, it can be difficult to diagnose errors due to incorrect initialization. Now the game crashes in this case, it seems to me that this is the correct behavior (otherwise we would not even know about the linked error).

Reply:
Fair point. I'm planning on rewriting the version checker either way, so this can be like this for now.